proto/buf/alpha/registry/v1alpha1/audit_event.proto

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+// AuditActorType is the type of actor that triggered certain audited event.
+enum AuditActorType {
+  // AUDIT_ACTOR_TYPE_UNSPECIFIED is the default actor type for audited events.
+  AUDIT_ACTOR_TYPE_UNSPECIFIED = 0;
+  // AUDIT_ACTOR_TYPE_USER is the actor type for when the audited event is
+  // triggered by a user. This is the most common case.
+  AUDIT_ACTOR_TYPE_USER = 1;
+}
+
+// AuditActor is the actor who triggered or executed the audited event.
+message AuditActor {
+  // Type of actor who caused the audited event.
+  AuditActorType type = 1;
+  // Id of the actor who eecuted the action.
+  string id = 2;
+}
+
+// AuditResourceType is the type of the resource that was affected by the
+// audited event.
+enum AuditResourceType {
+  // AUDIT_RESOURCE_TYPE_UNSPECIFIED is the default type for audited events.
+  AUDIT_RESOURCE_TYPE_UNSPECIFIED = 0;
+  // AUDIT_RESOURCE_TYPE_USER is the type for when the audited event affected a
+  // user.
+  AUDIT_RESOURCE_TYPE_USER = 1;
+  // AUDIT_RESOURCE_TYPE_ORGANIZATION is the type for when the audited event
+  // affected an organization.
+  AUDIT_RESOURCE_TYPE_ORGANIZATION = 2;
+  // AUDIT_RESOURCE_TYPE_ORGANIZATION_MEMBER is the type for when the audited
+  // event affected an organization member.
+  AUDIT_RESOURCE_TYPE_ORGANIZATION_MEMBER = 3;
+  // AUDIT_RESOURCE_TYPE_REPOSITORY is the type for when the audited event
+  // affected a repository.
+  AUDIT_RESOURCE_TYPE_REPOSITORY = 4;
+  // AUDIT_RESOURCE_TYPE_REPOSITORY_MEMBER is the type for when the audited
+  // event affected a repository member.
+  AUDIT_RESOURCE_TYPE_REPOSITORY_MEMBER = 5;
+  // AUDIT_RESOURCE_TYPE_REPOSITORY_COMMIT is the type for when the audited
+  // event affected a repository commit.
+  AUDIT_RESOURCE_TYPE_REPOSITORY_COMMIT = 6;
+  // AUDIT_RESOURCE_TYPE_PLUGIN is the type for when the audited event affected
+  // a plugin.
+  AUDIT_RESOURCE_TYPE_PLUGIN = 7;
+}
+
+// AuditResource is the affected resource by the audited event.
+message AuditResource {
+  // Type of resource that was affected by the audited event.
+  AuditResourceType type = 1;
+  // Id of the affected resource by the audited event.
+  string id = 2;
```

Comment:
assuming `id` is fine here; was there a reason for `name` in the design doc though? (just double-checking, and similar comment for `AuditActor`)

Reply:
I guess at that time I was drafting w/out fine grain detail. Now writing the actual message I see not much use on adding a `name`, since we might have resources or actors that don't have a `"name"` (think of a system admin at some point?). Actors right now [are always users](https://www.notion.so/bufbuild/Audited-Events-List-db1e266e05ce4d0db48638ec6091b9b2), and resources most of the times also have names (users, organizations, repos, plugins), except for members. Those cases, I'd say it's fine to use the usernames, but it seemed more appropriate to include that in the actual event payload.

I see it could be useful for [searching and filtering later](https://www.notion.so/bufbuild/Audit-Logging-e96b318c64224cd8b2a34887fcacc54a#42318760b882432eb08ff5ebce5cb01f) by names, so they could live in a specific column, but that's also possible with a fuzzy search.

I'm in doubt, so I preferred not to add it. wdyt?

Reply:
yeah, I doubt it's necessary too - just thought I'd double-check as I saw that in the doc. 👍